TST: DataFrame.hist() does not get along with matplotlib.pyplot.tight_layout()

ResidentMario opened the pull request:
- [ ] closes #9351
 - [ ] tests added / passed
 - [ ] passes ``git diff upstream/master | flake8 --diff``
 - [ ] whatsnew entry

ResidentMario committed:
Add unit test for #9351

ResidentMario commented:
Note: in addition to the #9351 unit test this PR also adds a `_skip_if_mpl_1` method to the test utilities.

jreback commented on pandas/util/testing.py:
```diff
@@ -286,6 +286,14 @@ def _skip_if_mpl_1_5():
+    v = matplotlib.__version__
+    if v < LooseVersion('2.0.0') or v[0] == '0' or v[0] == '1':
+        import pytest
+        pytest.skip("matplotlib 1.5")
```
matpotlib < 2.0.0

jreback commented on pandas/tests/plotting/test_hist_method.py:
```diff
@@ -238,6 +238,20 @@ def test_hist_layout(self):
+        df.plot.hist()
+
+        try:
+            self.plt.tight_layout()
```
you don't need the try/except (if it works it won't raise)

add a 1-line comment & the issue number as a comment

jreback reviewed (changes requested):
couple of small comments

jreback commented on pandas/util/testing.py:
```diff
@@ -286,6 +286,14 @@ def _skip_if_mpl_1_5():
 
 
+def _skip_if_mpl_1():
+    import matplotlib
```
actually you don't need this, instead use:

``
if self.mpl_ge_2_0_0:
``

as these are defined in the base class

ResidentMario replied:
`self.mpl_get_2_0_0` I assume?

jreback replied:
no take this function out

jreback replied:
yes that's fine. I realize this just raises ``AttributeError``, so that would be caught by the tests anyhow.

ResidentMario committed:
Tweaks.

ResidentMario commented:
I rewrote the test using `mpl_ge_2_0_0` and removed the aux method. `pytest` isn't a name in `test_hist_method`, and I don't see a generic skip in `testing.py`, so I've just made the test conditional, e.g. there's no explicit skip. Let me know if that's OK or not.

jreback commented on pandas/tests/plotting/test_hist_method.py:
```diff
@@ -238,6 +238,16 @@ def test_hist_layout(self):
+        if self.mpl_ge_2_0_0:
+            df = DataFrame(randn(100, 2))
+            df.plot.hist()
+            self.plt.tight_layout()
```
I think you need a ``_check_plot_works`` here

ResidentMario replied:
Would `_check_plot_works(df.hist); self.plt.tight_layout()` work? The method seems to be designed for testing parameter inputs, but `tight_layout` isn't a histogram parameter.

jreback replied:
yes that's fine. I realize this just raises ``AttributeError``, so that would be caught by the tests anyhow.

ResidentMario committed:
add _check_plot_works; rm aux method

jreback commented:
lgtm. ping on green.

jreback commented:
actually, pls add a whatsnew note as well (I guess in Bug Fixes). This is really a mpl compat issue, and there isn't much we can do about it with < 2.0.0, so best just to note this in the whatsnew as well so readers can react to this.

ResidentMario committed:
Add whatsnew entry.

jreback commented:
lgtm. thanks.

jreback commented:
thanks!
